lib/class-sp-sync-manager.php

```diff
@@ -166,6 +166,26 @@ public function get_posts( $args = array() ) {
 	 * @return bool
 	 */
 	public function do_index_loop() {
+
+		/**
+		 * Advanced Post Cache can periodically return the wrong value for
+		 * found_posts, which can cause the reindex operation to abort prematurely.
+		 * We will unhook it during cron to prevent this.
+		 */
+		if ( wp_doing_cron() && class_exists( 'Advanced_Post_Cache' ) ) {
+			global $advanced_post_cache_object;
+			if ( ! empty( $advanced_post_cache_object ) ) {
+				remove_filter(
+					'found_posts',
+					array( $advanced_post_cache_object, 'found_posts' )
+				);
+				remove_filter(
+					'found_posts_query',
+					array( $advanced_post_cache_object, 'found_posts_query' )
+				);
+			}
+		}
+
```

Comment:
This feels out of scope for this function. I think it would be better in another file, (perhaps a `compat.php` file), and we add an action here to hook into, e.g. `sp_pre_index_loop` or similar.

Reply:
Sure thing!